app/src/main/java/com/ataulm/artcollector/gallery/ui/GalleryActivity.kt

```diff
@@ -29,10 +32,9 @@ class GalleryActivity : AppCompatActivity() {
         val adapter = GalleryAdapter(glideRequestManager)
         recyclerView.adapter = adapter
         recyclerView.addItemDecoration(GallerySpacingItemDecoration(resources))
-
-        viewModel.gallery.observe(this, DataObserver<UiGallery> { gallery ->
-            adapter.submitList(gallery)
-        })
+        lifecycleScope.launch {
+            viewModel.pagedGallery().collectLatest { adapter.submitData(it) }
```

Comment:
while `UiGallery` was a `Collection<UiPainting>`, `pagedGallery()` returns a `Flow<PagingData<UiPainting>>`.

⚠️  Rather than the ViewModel emitting a stream of UiModels, there's now a two-layered approach where the ViewModel can emit new `PagingData<UiPainting>` (the flow) via the `Pager`, and the `Pager` essentially acts as a VM delegate: it's the one responsible for emitting state and also responding to requests for more data/refreshing which the VM would have been responsible for.

Feels weird but maybe less so when considered as a kind of delegate.